### PR TITLE
Fix Guacamole refresh token

### PR DESCRIPTION
**Pull request by anatbal**
# PR for issue #1110

## What is being addressed

Changes in Guacamole service to allow refresh token - by refactoring UserContext and adding update method

### CODE REVIEW

File: templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public static Map<String, Connection> getConnections(final AzureTREAuthenticated
         return connections;
     }
 
-    private static Map<String, GuacamoleConfiguration> getConfigurations(final AzureTREAuthenticatedUser user)
+    public static Map<String, GuacamoleConfiguration> getConfigurations(final AzureTREAuthenticatedUser user)
```
Comment (anatbal) [inline templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java:65]: Why public?

File: templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java [hunk 2]
```diff
@@ -57,6 +59,17 @@ public String getIdentifier() {
         return "azuretre";
     }
 
+    @Override
+    public AuthenticatedUser updateAuthenticatedUser(AuthenticatedUser authenticatedUser, Credentials credentials)
+        throws GuacamoleException {
+        LOGGER.info("updateAuthenticatedUser");
+        AuthenticatedUser updated = authenticateUser(credentials);
+
+        LOGGER.info("updateAuthenticatedUser - done");
```
Comment (anatbal) [inline templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java:68]: Do we still need this?

Comment (eladiw) [reply templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java:68]: no

File: templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java (hunk 1 above)
Comment (eladiw) [reply templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java:65]: so you can call it from the provider before you create the conntext

Comment (anatbal) [reply templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java:65]: But getConfigurations is only called inside the class

File: templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java (hunk 2 above)
Comment (anatbal) [reply templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java:68]: I am removing it then

File: templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java (hunk 1 above)
Comment (eladiw) [reply templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/connection/ConnectionService.java:65]: left overs. can revert

**Review by eladiw — APPROVED** [review]

**Review by marrobi — APPROVED** [review]
LGTM

File: templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java [hunk 3]
```diff
@@ -112,4 +127,14 @@ public UserContext getUserContext(final AuthenticatedUser authenticatedUser) thr
         }
         return null;
     }
+
+    @Override
+    public UserContext updateUserContext(UserContext context, AuthenticatedUser authenticatedUser,
+        Credentials credentials)
+        throws GuacamoleException {
+        LOGGER.info("Updating usercontext");
```
[the remark this answers is not in the document]
Comment (marrobi) [reply templates/workspace_services/guacamole/guacamole-server/guacamole-auth-azure/src/main/java/org/apache/guacamole/auth/azuretre/AzureTREAuthenticationProvider.java:135]: If its debugging, should it be debug?

**Comment by eladiw** [comment]
Good job

**Pull request merged**